src/state/axisSelectors.ts

```diff
@@ -607,11 +607,59 @@ export const selectAxisDomain = (
   return selectNumericalDomain(state, axisSettings, axisType, axisId);
 };
 
+const unknownScale: ParsedScaleReturn = {
+  scale: undefined,
+  realScaleType: undefined,
+};
+
+/**
+ * This returns an axis scale and type (linear, log, etc.) for the given axis.
+ * This is before applying the correct domain and range!
+ *
+ * That's on one hand silly, yes, but on the other hand also required
+ * because the final domain depends on nice ticks,
+ * and nice ticks depend on the scale type.
+ * With this shared dependency we cannot create the scale and set its domain in one step.
+ *
+ * Usually you would never want to use this selector, ever, outside of this circular dependency problem.
```

Comment:
Hmm, I wonder what our path away from this is then

Reply:
For now, no path away, it's fine and it can stay like this. An implementation detail, never exported.

If we choose to separate `parseScale` into two methods, one that decides on the `realScaleType` and second which actually creates the scale function/object then we might work around it. But then we get a new problem of transforming from `realScaleType` to scale function so I was in no hurry in doing that.